design-documents/graph-ql/coverage/catalog/compare-list.md

```diff
@@ -83,3 +103,25 @@ preconfigured at the backoffice.
   
 Guest compare list business logic not implemented yet. Additional development required.
 
+## Current limitations
+
+Existing table structure for compare list
+```
+catalog_compare_item
+------------------------------------------------------------------------------
+| catalog_compare_item_id | visitor_id | customer_id | product_id | store_id |
+==============================================================================
+```
+The visitor_id (non nullable) is tied to session \Magento\Customer\Model\Visitor in Luma. GraphQl layer should not be tied to Luma's session. So Compare-lists via GraphQl will have to differ/deviate from Luma.
+
+This dependency can be solved by managing the compare list state in a new table
+```
+catalog_compare_list
```

Comment:
Will only have the following fields: `list_id`, `visitor_id`. Not sure we can avoid having here `visitor_id`, we need somehow to make sure that `catalog_compare_item`.`visitor_id` is the same for all items related to a list.

Reply:
visitor_id is generated based on Luma's session. so we cannot mock it. But it can be 0, which will imply the record is not created via Luma. And I see your point.We limit the new table to have two columns list_id, additional_data (json) and we can refer it back to the original table.